Clarify role of prefix in completers

evanunderscore opened the pull request:
Fixes #139

evanunderscore commented:
@kislyuk I've altered some of the documentation and examples to remove the use of the prefix as a filter. I've also change the actual completers to do the same and be generally simpler and more lazy. Happy to undo some of these changes if you don't think they're necessary.

kislyuk commented on docs/examples/describe_github_user.py:
```diff
@@ -2,9 +2,9 @@
+def github_org_members(parsed_args, **kwargs):
     resource = "https://api.github.com/orgs/{org}/members".format(org=parsed_args.organization)
-    return (member['login'] for member in requests.get(resource).json() if member['login'].startswith(prefix))
+    return (member['login'] for member in requests.get(resource).json())
```
I want to retain this in the example. Please don't remove it.

kislyuk commented on README.rst:
```diff
@@ -139,9 +140,9 @@ organization and complete their names, then prints the member description:
+    def github_org_members(parsed_args, **kwargs):
         resource = "https://api.github.com/orgs/{org}/members".format(org=parsed_args.organization)
-        return (member['login'] for member in requests.get(resource).json() if member['login'].startswith(prefix))
+        return (member['login'] for member in requests.get(resource).json())
```
Same here, let's retain this.

kislyuk commented:
Looks OK modulo the two comments, let's keep the examples.

evanunderscore commented:
I'll revert those two.

Perhaps if we added another example that uses the prefix they could be removed again? Something like #94 looks like a good candidate for this, e.g.:

```
if '=' not in prefix:
    return get_names()
name, _ = prefix.split('=')
return (f'{name}={v}' for v in get_versions(name))
```

As it stands the example gives the impression that filtering is a useful thing for a completer to do.

evanunderscore commented:
I've reverted the changes to the example script. I'll merge this in. Thanks for the review.
